csharp/ql/test/library-tests/frameworks/EntityFramework/EntityFramework.cs

```diff
@@ -0,0 +1,65 @@
+// semmle-extractor-options: /r:System.Data.dll /r:System.ComponentModel.Primitives.dll /r:System.ComponentModel.TypeConverter.dll ${testdir}/../../../resources/stubs/EntityFramework.cs ${testdir}/../../../resources/stubs/System.Data.cs /r:System.ComponentModel.TypeConverter.dll /r:System.Data.Common.dll
+
+using System.Data.Entity;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Data;
+using System.Data.Common;
+
+namespace EFTests
+{
+    class Person
+    {
+        public int Id { get; set; }
+        public string Name { get; set; }
+
+        [NotMapped]
+        public int Age { get; set; }
+    }
+
+    class MyContext : DbContext
+    {
+        DbSet<Person> person { get; set; }
+
+        void FlowSources()
+        {
+            var p = new Person();
+            var id = p.Id;
+            var name = p.Name;
+            var age = p.Age;
+        }
+
+        DbCommand command;
+
+        async void SqlSinks()
+        {
+            // System.Data.Common.DbCommand.set_CommandText
+            command.CommandText = "";
+            
+            // System.Data.SqlClient.SqlCommand.SqlCommand
+            new System.Data.SqlClient.SqlCommand("");
+            
+            this.Database.ExecuteSqlCommand("");
+            await this.Database.ExecuteSqlCommandAsync("");
+        }
+
+        void TestDataFlow()
+        {
+            string taintSource = "tainted";
+
+            // Tainted via database, even though technically there were no reads or writes to the database in this particular case.
```

Comment:
Do you think we could perhaps use the `TaintedMember` class that you introduced on https://github.com/Semmle/ql/pull/847/files#diff-090d1ac1f70ea89b57dab64ece1357a3R23 to get more accuracy?

I am thinking something like:

- `Database.SqlQuery` (and others) are remote taint sources.
- There is a `TaintedMember` taint step from mapped objects to mapped properties, and vice versa (this would require a new abstraction).
- There is a jump step from all DB writes of a given type, to all DB reads of the same type.